examples/all-clusters-app/esp32/main/StatusScreen.cpp

```diff
@@ -0,0 +1,189 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ * @file StatusScreen.cpp
+ *
+ * Screen which displays device state and status:
+ *      - Fabric Index : none | #
+ *      - Fabric ID : none | #
+ *      - Node ID : none | #
+ *      - UDP Port : #
+ *      - IPv4 : none | addr
+ *      - IPv6 LL : none | addr
+ *      - IPv6 ULA : none | addr
+ */
+
+#include "StatusScreen.h"
+
+#if CONFIG_HAVE_DISPLAY
+
+#include "esp_log.h"
+#include "esp_system.h"
+#include "esp_wifi.h"
+#include "freertos/FreeRTOS.h"
+#include "freertos/task.h"
+
+#include <lib/support/CodeUtils.h>
+#include <platform/CHIPDeviceLayer.h>
+
+#include <app/server/Server.h>
+#include <lib/core/DataModelTypes.h>
+
+#define MAX_LENGTH_SMALL_FONT 30
+
+using namespace chip;
+
+class StatusListModel : public ListScreen::Model
+{
+public:
+    enum
+    {
+        kStatusItemFabricIndex,
+        kStatusItemFabricId,
+        kStatusItemNodeId,
+        kStatusItemPort,
+        kStatusItemIp4,
+        kStatusItemIp6LinkLocal,
+        kStatusItemIp6Ula,
+    };
+
+    StatusListModel()
+    {
+        options.emplace_back("FabricIndex : none");
+        options.emplace_back("FabricID : none");
+        options.emplace_back("NodeID : none");
+        options.emplace_back("UDP port : " + std::to_string(CHIP_PORT));
+        options.emplace_back("IPv4 : none");
+        options.emplace_back("IPv6 LL : none");
+        options.emplace_back("IPv6 ULA : none");
+    }
+
+    virtual std::string GetTitle() { return "Status"; }
+
+    virtual int GetItemCount() { return options.size(); }
+    virtual std::string GetItemText(int i)
+    {
+        std::string itemString = options.at(i);
+        switch (i)
+        {
+        case kStatusItemFabricIndex: {
+            for (const auto & fb : Server::GetInstance().GetFabricTable())
```

Comment:
@turon For what it's worth, this has issues similar to https://github.com/project-chip/connectedhomeip/issues/15065

We should probably really queue this stuff over to the Matter thread and then block until it runs or something....